lua/compe/completion.lua

```diff
@@ -228,7 +228,9 @@ Completion._show = function(start_offset, items)
   vim.schedule(function()
     local completeopt = vim.o.completeopt
     vim.cmd('set completeopt=menu,menuone,noselect')
-    vim.call('complete', start_offset, items)
+    if not vim.tbl_isempty(items) then
+      vim.call('complete', start_offset, items)
+    end
```

Comment:
You should use `if not (vim.call('pumvisible') == 0 and #items == 0) then`.

vim needs `call complete(1, [])` to close popup menu. so we should check `pumvisible` in here.

Reply:
Yeah! I didn't know about it! I was observing some abrupt behavior but I was
not sure of it. Thanks! I just added a commit regarding it. The only condition we would get ``Pattern not found`` now is when we type something that didn't match from popup menu suggestions. I don't know if we can also remove it.

Reply:
>The only condition we would get Pattern not found now is when we type something that didn't match from popup menu suggestions.
Yes. It is expected behavior.

You can set `set shortmess+=c` to avoid it.

Reply:
Yeah! That was so lame of me :sweat_smile:. I should have read help docs. Thanks! for help.